experimenter/evaluate_pipeline_new.py

```diff
@@ -0,0 +1,282 @@
+        data_random_seed=data_random_seed)
+
+    save_pipeline_run_to_d3m_db(output_run_path)
+
+def evaluate_pipeline_via_d3m_cli(pipeline: str,
+    problem: str,
+    input: str,
+    output_run: str,
+    data_random_seed: int,
+    data_params: List[Tuple[str,Any]] = None,
+    data_pipeline: str = K_FOLD_TABULAR_SPLIT_PIPELINE_ID,
+    scoring_pipeline: str = SCORING_PIPELINE_ID,
+    input_run: str = None,
+    metric: str = None,
+    scoring_params: List[Tuple[str,Any]] = None,
+    scores: str = None,
+    scoring_random_seed: int = None,
+    data_split_file: str = None):
+    """ 
+    Evaluate pipeline on problem using d3m's runtime cli. 
+    Wrapper function to execute d3m's runtime cli 'evaluate' command.
+    Arguments mirror the same arguments using the cli.
+
+    Parameters
+    ----------
+    pipeline : path_like or uuid4 str
+        path to pipeline doc or pipeline ID
+    problem : path_like str
+        path to problem doc
+    input : path_like str
+        path to input full data
+    output_run : path_like str or '-'
+        path where pipeline_run doc
+        will be saved.
+        use '-' for stdin
+    data_random_seed : int
+        random seed to use for
+        data preparation
+    data_params : list of tuples, optional
+        hyper-parameter names and values
+        for data preparation.
+        None by default
+    data_pipeline : path_like str or uuid4 str, optional
+        path to data preparation pipeline file
+        or pipeline ID.
+        K_FOLD_TABULAR_SPLIT_PIPELINE_ID by default
+    scoring_pipeline : path_like str or uuid4 str, optional
+        path to scoring pipeline file
+        or pipeline ID.
+        SCORING_PIPELINE_ID by default
+    input_run : path_like str or '-', optional
+        path to pipeline_run file
+        with configuration.
+        use '-' for stdin.
+        None by default
+    metric : str, optional
+        metric to use.
+        Metric from problem by default
+    scoring_params : list of tuples, optional
+        hyper-parameter names and values
+        for scoring pipeline.
+        None by default
+    scores : path_like str, optional
+        path to save scores.
+        None by default
+    scoring_random_seed : int, optional
+        random seed to use for scoring.
+        None by default
+    data_split_file : path_like str, optional
+        reads the split file and populates
+        "primary_index_values" hyper-parameter
+        for data preparation pipeline with values
+        from the "d3mIndex" column corresponding
+        to the test data.
+        use '-' for stdin.
+        None by default
+
+    Return:
+    -------
+    None
+    
+    Raises:
+    -------
+    TypeError
+        when parameter value has 
+        incorrect type
+    ValueError
+        when parameter value is
+        invalid
+    """
+    args = ['d3m', 'runtime', 'evaluate']
+
+    if (not isinstance(pipeline, str)):
```

Comment:
All this validation is not necessary. We can use mypy to check that the right types are being passed in.

Reply:
Okay, I'll make the changes.